reactor-netty-core/src/main/java/reactor/netty/transport/ServerTransport.java

```diff
@@ -532,21 +535,31 @@ public void disposeNow(Duration timeout) {
 			}
 			dispose();
 			Mono<Void> terminateSignals = Mono.empty();
-			if (config.channelGroup != null) {
-				List<Mono<Void>> channels = new ArrayList<>();
+			if (config.channelGroup != null && config.channelGroup.size() > 0) {
+				HashMap<Channel, List<Mono<Void>>> channelsToMono = new HashMap<>(config.channelGroup.size());
```

Comment:
https://bugs.openjdk.org/browse/JDK-8186958 
> It seems that people normally assume that the constructor HashMap(int) requires the expected size as the argument.
In fact, the argument is the initial capacity, which should be bigger than the expected size. As a result, unnecessary reallocations can occur while the maps are filled.

Reply:
@robotmrv What is your proposal?

Reply:
I've just noticed possible misuse of the HashMap constructor.  
Probably it could improved the same way as in jdk or guava analog via utility method 
https://github.com/openjdk/jdk/pull/7928/files#diff-9837dc8a9161163b6a1489098920d47c7fb62f19d9f929c1447f3c1c9bf7aa68R2581

or 
https://github.com/google/guava/blob/master/guava/src/com/google/common/collect/Maps.java#L277

Reply:
I am ok 
but quick search shows that there are several similar cases   
https://github.com/reactor/reactor-netty/blob/143191ae6db99c07e509cb62b60e362b771c2fa5/reactor-netty-core/src/main/java/reactor/netty/transport/TransportConfig.java#L354  
https://github.com/reactor/reactor-netty/blob/058159592584fcc3a66c32b48a5bfe6a8fe09297/reactor-netty-http/src/main/java/reactor/netty/http/server/HttpPredicate.java#L347
https://github.com/reactor/reactor-netty/blob/2ea7d622242f8750b70844c0a69a49446e8872f1/reactor-netty-http/src/main/java/reactor/netty/http/server/HttpServerBind.java#L44

probably it is a good idea to evaluate whether they require the same fix.  
If so - create another PR and move this logic to the utility method

Reply:
here it is #2739